docs/Grammar.md

```diff
@@ -0,0 +1,83 @@
+# Pact Grammar
+
+Here ye lies the pact bnf, based on [the current spec](https://pact-language.readthedocs.io/en/stable/pact-reference.html).
+
+Note: happy is more efficient in parsing left-recursive s-expressions, so the grammar is being stated that way
+(see: https://www.haskell.org/happy/doc/html/sec-sequences.html).
+
+```ebnf
+(* A program is a collection of modules and expressions *)
+program        ::= top_level_list
+top_level_list ::= module top_level_list | 
+                     expr top_level_list | ε
+
+module ::= '(' 'module' ident kset doc_or_meta decl_list ')'
+
+(* top level declarations *)
+decl_list ::= imports decls
+imports ::= '(' use ')' { '(' use ')' }
+decls ::=  '(' decl ')' { '(' decl ')' }
+decl      ::= defun | defschema | defconst | defcap | defpact |
+              deftable | bless
+
+(* Todo: what is implements? *)
```

Comment:
Implements is detailed in these docs: https://github.com/kadena-io/pact/pull/531

Basically, a module `implements` an interface in the sense that it must provide a body for all function signatures in the interface. When this is done, its models are `<>`'d in the module body/defun definition and you're good to go. 

Think ML style modules which concat boolean expressions given by @model